### PR TITLE
Add className prop to top level components.

### PR DESCRIPTION
**Pull request by samhogg**
First cut, this follows discussion in #73.

Not sure if updating docs to add `className` entries for the API is really necessary? Happy to discuss 😄

### CODE REVIEW

File: src/legends/continuous-color-legend.js [hunk 1]
```diff
@@ -39,7 +39,8 @@ const propTypes = {
     React.PropTypes.string
   ]).isRequired,
   height: React.PropTypes.number,
-  width: React.PropTypes.number
+  width: React.PropTypes.number,
+  className: React.PropTypes.string
```
Comment (mcnuttandrew) [inline src/legends/continuous-color-legend.js]: could you alphabetize these props?

File: src/legends/continuous-color-legend.js [hunk 2]
```diff
@@ -48,14 +49,14 @@ const defaultProps = {
 };
 
 function ContinuousColorLegend({startColor, midColor, endColor, startTitle,
-  midTitle, endTitle, height, width}) {
+  midTitle, endTitle, height, width, className = ''}) {
```
Comment (mcnuttandrew) [inline src/legends/continuous-color-legend.js]: probably should do this with default props instead

File: src/legends/continuous-size-legend.js [hunk 3]
```diff
@@ -33,7 +33,8 @@ const propTypes = {
   endSize: React.PropTypes.number,
   circlesTotal: React.PropTypes.number,
   height: React.PropTypes.number,
-  width: React.PropTypes.number
+  width: React.PropTypes.number,
+  className: React.PropTypes.string
```
Comment (mcnuttandrew) [inline src/legends/continuous-size-legend.js]: alphabetize

File: src/legends/continuous-size-legend.js [hunk 4]
```diff
@@ -43,7 +44,7 @@ const defaultProps = {
 };
 
 function ContinuousSizeLegend({startTitle, endTitle, startSize, endSize,
-  circlesTotal, height, width}) {
+  circlesTotal, height, width, className = ''}) {
```
Comment (mcnuttandrew) [inline src/legends/continuous-size-legend.js]: default props

File: src/legends/searchable-discrete-color-legend.js [hunk 5]
```diff
@@ -46,14 +46,15 @@ function SearchableDiscreteColorLegend(props) {
     searchText,
     searchPlaceholder,
     width,
-    height
+    height,
+    className = ''
```
Comment (mcnuttandrew) [inline src/legends/searchable-discrete-color-legend.js]: alphabetize and default props

File: src/radial-chart/index.js [hunk 6]
```diff
@@ -77,6 +77,7 @@ class RadialChart extends React.Component {
   static get propTypes() {
     return {
       animation: AnimationPropType,
```
Comment (mcnuttandrew) [inline src/radial-chart/index.js]: alphabetize

File: src/radial-chart/index.js [hunk 7]
```diff
@@ -295,6 +296,7 @@ class RadialChart extends React.Component {
   render() {
     const {
       animation,
+      className = '',
```
Comment (mcnuttandrew) [inline src/radial-chart/index.js]: default props

File: src/treemap/index.js [hunk 8]
```diff
@@ -83,6 +83,7 @@ class Treemap extends React.Component {
     return {
       animation: AnimationPropType,
```
Comment (mcnuttandrew) [inline src/treemap/index.js]: alphabetize

**Comment by samhogg** [comment]
@mcnuttandrew  thanks - will go through and fix those bits up - lots weren't alphabetized anyway so wasn't quite sure.

Fix incoming.

File: src/plot/xy-plot.js [hunk 9]
```diff
@@ -305,6 +310,10 @@ class XYPlot extends React.Component {
   }
 }
 
+XYPlot.defaultProps = {
```
Comment (samhogg) [inline src/plot/xy-plot.js]: Sure, will fix.

Comment (samhogg) [inline src/plot/xy-plot.js]: @Apercu done 😄

**Pull request merged**